File: Python/003-LongestSubstringWithoutRepeatingCharacters/SolutionByViolent.py
# !/usr/bin/env Python3
# -*- coding: utf-8 -*-

#  @Author: Frenude 
#  @Description:
#  @Date: Created in 01 06,2021
#  @Modified By:

class SolutionByViolent:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if not len(s):
            return 0
        max_l = 0
        for i in range(len(s)):
            str_set = set()
            l = 0
            for j in range(i,len(s)):
                if s[j] in str_set:
                    break
                str_set.add(s[j])
                l += 1
            max_l = max(l,max_l)
        return max_l
if __name__ == '__main__':
    s = "pwwkew"
    so = SolutionByViolent()
    z = so.lengthOfLongestSubstring(s)
    print(z)

File: Java/003-LongestSubstringWithoutRepeatingCharacters/SolutionByViolent.java
import java.util.HashMap;
import java.util.HashSet;

/**
 * @Author: Frenude
 * @Description: //TODO
 * @Date: Created in 01 05,2021
 * @Modified By:
 **/


public class SolutionByViolent {

    public static void main(String[] args) {
        String s = "pwwkew";
        SolutionByViolent solution = new SolutionByViolent();
        int i = solution.lengthOfLongestSubstring(s);
        System.out.println(i);
    }


    public int lengthOfLongestSubstring(String s) {
        if (s == null && s.length() == 0) {
            return 0;
        }
        int max_l = 0;
        HashSet<Character> set = new HashSet<>();
        for (int i = 0; i < s.length(); i++) {
            int l = 0;
            for (int j = i; j < s.length(); j++) {
                if (set.contains(s.charAt(j))) {
                    break;
                }
                set.add(s.charAt(j));
                l++;
            }
            max_l = Math.max(l, max_l);
            set.clear();
        }
        return max_l;
    }

}
